src/dmd/clone.d

```diff
@@ -765,7 +765,10 @@ extern (C++) FuncDeclaration buildXtoHash(StructDeclaration sd, Scope* sc)
     const(char)* code =
         "size_t h = 0;" ~
         "foreach (i, T; typeof(p.tupleof))" ~
-        "    h = h * 33 + typeid(T).getHash(cast(const void*)&p.tupleof[i]);" ~
+        "    static if(is(T* : const(Object)*)) " ~
```

Comment:
The need for comparing pointers is slightly subtle.

Reply:
Checking for a class and using `__traits(getLinkage)` to exclude C++ and Objective-C classes might have been more obvious.

Reply:
Really, this is never seen by anyone. So I don't know that it matters what technique is used. I'm hoping that at some point instead of adjusting this, we simply move the function generation into an object.d template.

I'm also hoping that the underlying issue is fixed, and this workaround becomes moot.